dartagnan/src/main/java/com/dat3m/dartagnan/utils/options/BaseOptions.java

```diff
@@ -45,6 +92,48 @@ public void parse(String[] args) throws ParseException, RuntimeException {
         if(cmd.hasOption("target")) {
             target = Arch.get(cmd.getOptionValue("target"));
         }
+        
+        scope = cmd.hasOption(METHOD_OPTION) ? ScopeTypes.fromString(cmd.getOptionValue(METHOD_OPTION)) : INCREMENTAL;
+        if(!supported_scope.contains(scope)) {
+            throw new UnsupportedOperationException("Unrecognized solver method: " + scope);        		
+        }
+
+        smtsolver = Z3;
+        if(cmd.hasOption(SMTSOLVER_OPTION)) {
+			
+        	if(!supported_smtsolvers.contains(cmd.getOptionValue(SMTSOLVER_OPTION))) {
+                throw new UnsupportedOperationException("Unrecognized SMT solver: " + cmd.getOptionValue(SMTSOLVER_OPTION));        		
+            }
+
+			switch(cmd.getOptionValue(SMTSOLVER_OPTION)) {
+				case "mathsat5":
+					smtsolver = MATHSAT5;
+					break;
+				case "smtinterpol":
+					smtsolver = SMTINTERPOL;
+					break;
+				case "princess":
+					smtsolver = PRINCESS;
+					break;
+				case "boolector":
+					smtsolver = BOOLECTOR;
+					break;
+				case "cvc4":
+					smtsolver = CVC4;
+					break;
+				case "yices2":
+					smtsolver = YICES2;
+					break;
```

Comment:
adding a "z3" and "unspecified" case that both select z3 as solver might be more readable. A default case could throw the exception. This way you avoid all Ifs. Something like:
```
String solverOption = cmd.getOptionValue(SMTSOLVER_OPTION, "unspecified");
switch (solverOption) {
  // cases
  // ...
  case "z3":
  case "unspecified":
     smtsolver = Z3;
     break;
  case default: 
     // throw exception
}